ext/phar/phar_object.stub.php

```diff
@@ -12,138 +12,138 @@ public function __construct(string $filename, int $flags = FilesystemIterator::S
 
     public function __destruct() {}
 
-    /** @return void */
-    public function addEmptyDir(string $directory) {}
+    /** @tentative-return-type */
+    public function addEmptyDir(string $directory): void {}
 
-    /** @return void */
-    public function addFile(string $filename, ?string $localName = null) {}
+    /** @tentative-return-type */
+    public function addFile(string $filename, ?string $localName = null): void {}
 
-    /** @return void */
-    public function addFromString(string $localName, string $contents) {}
+    /** @tentative-return-type */
+    public function addFromString(string $localName, string $contents): void {}
 
-    /** @return array|false */
-    public function buildFromDirectory(string $directory, string $pattern = "") {}
+    /** @tentative-return-type */
+    public function buildFromDirectory(string $directory, string $pattern = ""): array|false {} // TODO Remove false?
 
-    /** @return array|false */
-    public function buildFromIterator(Traversable $iterator, ?string $baseDirectory = null) {}
+    /** @tentative-return-type */
+    public function buildFromIterator(Traversable $iterator, ?string $baseDirectory = null): array|false {} // TODO Remove false?
 
-    /** @return void */
-    public function compressFiles(int $compression) {}
+    /** @tentative-return-type */
+    public function compressFiles(int $compression): void {}
 
     /** @return bool */
-    public function decompressFiles() {}
+    public function decompressFiles() {} // TODO make return type void
```

Comment:
A very promising solution came to my mind overnight for dealing with these todos in the future: we could check if the return value is used via `USED_RET()`, and if that's the case, we could emit a deprecation notice such as "Phar::decompressFiles() will have a void return value in the future instead of true" (hopefully we can come up with a better wording until then :) )

Reply:
It's not possible to determine whether the return value is used for calls using `call_user_func` or similar.